### PR TITLE
ncm-spma: Support yum gpgkey/gpgcakey/repo_gpgcheck repository settings

### LINKED ISSUE #717: ncm-spma: yum schema does not support gpgkey option
The yum schema has support for the gpgcheck option to check GPG signatures on packages but does not support the gpgkey option to allow per repository gpg keys - we sign our locally built rpms and have a seperate repository for them and it would be good to be able to specify the key in quattor

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #717

### CODE REVIEW

**Comment by stdweird** [comment]
@jrha you can change the milestone :smile:

**Comment by jrha** [comment]
gpgcakey = gpg:cake:y?

File: ncm-spma/src/main/pan/components/spma/software.pan [hunk 1]
```diff
@@ -7,6 +7,8 @@ declaration template components/spma/software;
 include 'pan/types';
 include 'components/spma/functions';
 
+type software_repository_url = string with match(SELF, '^(file|https?|ftp)://');
```
Comment (jrha) [inline ncm-spma/src/main/pan/components/spma/software.pan]: What about one of the URI types from `pan/types`?

Comment (stdweird) [inline ncm-spma/src/main/pan/components/spma/software.pan]: i saw that, but have no clue what ips wants.

Comment (jrha) [inline ncm-spma/src/main/pan/components/spma/software.pan]: @msmark?

**Comment by jrha** [comment]
Good enough for now, will open an issue to think about the URL type.

**Pull request merged**